src/cnaas_nms/api/device.py

```diff
@@ -302,24 +304,57 @@ def delete(self, device_id):
 
     @login_required
     @device_api.expect(device_model)
-    def put(self, device_id):
+    def put(self, device_id: int):
         """Modify device from ID"""
         json_data = request.get_json()
         with sqla_session() as session:  # type: ignore
             dev: Optional[Device] = session.query(Device).filter(Device.id == device_id).one_or_none()
-
             if not dev:
                 return empty_result(status="error", data=f"No device with id {device_id}"), 404
 
             dev_prev_state: DeviceState = dev.state
+            current_hostname: str = dev.hostname
+
+            is_name_change_request = "hostname" in json_data
+            name_change_ok = False
+            if is_name_change_request:
+                new_hostname = json_data["hostname"]
+                old_has_device_settings = has_hostname_specific_settings(current_hostname)
+                new_has_device_settings = has_hostname_specific_settings(new_hostname)
+                name_change_ok = new_has_device_settings or not old_has_device_settings
+
+            if is_name_change_request and not name_change_ok:
+                msg = f"Hostname change would result in loss of configuration {current_hostname}"
```

Comment:
Hostname change would result in loss of configuration ... becase of device specific settings in git repository